runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -272,7 +272,14 @@ public TerminalNode getToken(int ttype, int i) {
 
 	@Override
 	public Interval getSourceInterval() {
-		if ( start==null || stop==null ) return Interval.INVALID;
+		if ( stop==null ) {
+			if ( start!=null ) {
+				return Interval.of(start.getTokenIndex(), start.getTokenIndex()-1); // empty
+			}
+			else {
+				return Interval.of(0, -1);
```

Comment:
This should be `Interval.INVALID` since we really have no idea what the source interval is.